<?xml version="1.0"?>
<launch>

  <arg name="output" default="screen"/>
  <arg name="sim" default="true"/>
  <arg name="gui" default="false"/>
  <arg name="map_base_name" default="current"/>
  <arg name="map_base_name_sim" default="map_leipzig1_simuliert"/>
  <arg name="generate_costs" default="false"/>


  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="$(arg output)">

    <param name="controller_frequency" type="double" value="10.0" />
    <param name="planner_frequency" type="double" value="0.0" />
    <param name="oscillation_timeout" type="double" value="10" />
    <param name="base_local_planner" type="string" value="dwa_local_planner/DWAPlannerROS" />
    <param name="base_global_planner" type="string" value="global_planner/GlobalPlanner" />
    <param name="global_planner/use_dijkstra" type="bool" value="true"/>

   <!-- <rosparam file="$(find navigation_exercise)/config/move_base_params.yaml" command="load" ns="move_base" />-->

    <rosparam file="$(find navigation_exercise)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find navigation_exercise)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find navigation_exercise)/config/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find navigation_exercise)/config/global_costmap_params.yaml" command="load" />
    <!--rosparam file="$(find navigation_exercise)/config/dwa_local_planner_params_forward.yaml" command="load" ns="DWAPlannerROS" /-->
    <rosparam file="$(find navigation_exercise)/config/dwa_local_planner_params.yaml" command="load" ns="DWAPlannerROS" />
    <rosparam file="$(find navigation_exercise)/config/global_planner.yaml" command="load" ns="global_planner" />


    <remap from="/scan_filtered" to="/scan" if="$(arg sim)"/>
    <remap from="/scan_back_filtered" to="/scan_back" if="$(arg sim)"/>
    <remap from="odom" to="base_odometry/odom" if="$(arg generate_costs)"/>

  </node>


</launch>
